### DISCUSSION TITLE
Is there a way to read a tag by specifying the instanceID?

### DISCUSSION THREAD

**Discussion by bubulindo** [ottowayi/pycomm3#229] [Q&A]
Hello, 

I'm looking at understanding how the new Extended Properties work and to change them online and one of these days when exporting alarm texts noticed that the text was marked as an InstanceID followed by @.@description and wondered if there's a way to read a tag by instance Id (which hopefully would allow me to concat the string above. 

Anyone?

**Comment by ottowayi** [comment]
There isn't a way to read a tag by instance id directly aka `plc.read(1234)`, but they are used in the underlying request when supported.  Like firmware version >= 21, not a Micro800, not a program-scoped tag, etc determine if the driver uses an instance id or symbolic name when it builds the request.  But, you do have access to them via `tags` property on the driver.  For example, here are some of the tags in my PLC I use for testing:

```python
with LogixDriver('192.168.1.236') as plc:
    for tag, info in plc.tags.items():
        print(f"{tag}: {info['instance_id']}")
```

Output:
```shell
write_sint_min: 119
write_dint_2d_ary1: 398
write_int2: 496
write_str480_2: 543
read_str_symbols: 868
read_lint_min: 989
read_dint_max: 1856
write_lint2: 2087
Local:4:O: 3815
Local:3:I: 6353
write_lint_ary1: 7022
write_dint_3d_ary1: 7065
...
```
